tester/src/test/scala/spinal/tester/scalatest/SpinalSimWishboneSlaveFactoryTester.scala

```diff
@@ -43,55 +46,66 @@ class SpinalSimWishboneSlaveFactoryTester extends SpinalAnyFunSuite{
       for(ref <- List(
         WishboneTransaction(10 * wordInc, 100), WishboneTransaction(0, 1),
         WishboneTransaction(10 * wordInc, 100), WishboneTransaction(0, 1),
-        WishboneTransaction(10 * wordInc, 100), WishboneTransaction(0, 2)
+        WishboneTransaction(10 * wordInc, 100), WishboneTransaction(0, 2),
+        WishboneTransaction(21 * wordInc, 1),
+        WishboneTransaction(21 * wordInc, 1),
+        WishboneTransaction(21 * wordInc, 42)
       )) {
         scoreboard.pushRef(ref)
       }
 
-      WishboneMonitor(dut.io.bus, dut.clockDomain){ bus =>
-           scoreboard.pushDut(WishboneTransaction.sampleAsMaster(bus))
-      }
+      new WishboneMonitor(dut.io.bus, dut.clockDomain).addResponseCallback(
+        (bus, transaction, we) => {
+          scoreboard.pushDut(transaction.copy(data = bus.DAT_MISO.toBigInt))
+        }
+      )
+
+      // Make sure that all configurations exhibit proper latencies.
+      val expectedResponseTime = if(regFeedback) 10 else 0
+      val busStatus = WishboneStatus(dut.io.bus)
+      var responseTime : Option[Long] = None
+      dut.clockDomain.onSamplings({
+        if(busStatus.isCycle) {
+          // Start the clock as soon as we see a request
+          if (busStatus.masterHasRequest && responseTime.isEmpty) {
+            responseTime = Some(simTime())
+          }
+
+          // Measure until ACK time.
+          if(dut.io.bus.ACK.toBoolean) {
+            val elapsedTime = simTime() - responseTime.get
+            //assert(expectedResponseTime == elapsedTime)
+
+            // For the pipeline case; if STB is assert it means we got a request now
+            responseTime = if(conf.isPipelined && dut.io.bus.STB.toBoolean && regFeedback) Some(simTime()) else None
+          }
+        }
+      })
 
       val dri = new WishboneDriver(dut.io.bus, dut.clockDomain)
       dri.drive(scala.collection.immutable.Seq(WishboneTransaction(10*wordInc), WishboneTransaction(0)), we = false)
       dri.drive(scala.collection.immutable.Seq(WishboneTransaction(10*wordInc, 200), WishboneTransaction(0, 2)), we = true)
       dri.drive(scala.collection.immutable.Seq(WishboneTransaction(10*wordInc), WishboneTransaction(0)), we = false)
+      dri.drive(scala.collection.immutable.Seq(WishboneTransaction(21*wordInc, 0)), we = false)
```

Comment:
simply having read/write methods would make this a bunch simpler - but I guess not worth it since it works as is...